packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -294,11 +294,15 @@ const helperBehavior = {
     // callback that may not fire until much later.
     // Callers shouldn't expect a response from this function.
     const { priceAuthority } = state;
-    state.outstandingQuote = await E(priceAuthority).mutableQuoteWhenLT(
+    trace('posting quote request', triggerPoint);
+
+    state.outstandingQuote = E(priceAuthority).mutableQuoteWhenLT(
```

Comment:
the comment above is no longer accurate. maybe:
> There are two promises here to get the quote. The first gets a mutableQuote object from the PriceAuthority but it's not awaited. …

Alternately, this would be the second time a change in an important design comment was hidden from code review diff. I think that's because it's not obvious from the code flow that there's some block that's being documented. Consider encapsulating this logic into a helper function like:
```
const quote = await updatedQuote(priceAuthority, highestDebtRatio.denominator, triggerPoint);
```
And the jsdoc on that function can explain the design.

Reply:
that woudl have the same bug as the `await` bug this is fixing.  I'll add comments here.

Reply:
Then I don't understand the bug. How will 24ea7c83f fail?

That change doesn't fail any tests. Does the bug have a regression test?

Reply:
That's further obscuring the interlock around `outstandingQuote`: 
```
if (!outstandingQuote) {
  ...return...
}
outstandingQuote = somethingNonNull;
```
The key is that MUST be an atomic block without an `await`. Putting the `await` before the `somethingNonNull` opens a race that hid a test failure. I'm considering refactoring it a different way so that the atomicity can't be compromised.

Reply:
New comment looks great. thanks.